src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/repository/impl/ords/JJDisputeRepositoryImpl.java

```diff
@@ -80,8 +91,65 @@ public JJDispute saveAndFlush(JJDispute entity) {
 	}
 
 	@Override
-	public void setStatus(String ticketNumber, JJDisputeStatus jjDisputeStatus, String userName) {
-		throw new NotYetImplementedException();
+	public void setStatus(String ticketNumber, JJDisputeStatus jjDisputeStatus, String userName, String partId, Long courtAppearanceId) {
+		try {
+			JJDispute jjDispute = map(jjDisputeApi.v1JjDisputeGet(ticketNumber));
+			// For some reason ORDS returns a valid object but with null fields if no object is found.
+			if (jjDispute != null && !StringUtils.isBlank(jjDispute.getTicketNumber())) {
+
+				String seizedYn = null;
+				String adjudicatorPartId = null;
+				String aattCd = null;
+				String dattCd = null;
+				String staffPartId = null;
+
+				if (!CollectionUtils.isEmpty(jjDispute.getJjDisputeCourtAppearanceRoPs()) && courtAppearanceId != null && partId != null) {
```

Comment:
May I ask why partId is required here when fetching a courtAppearance by courtAppearanceId?  Is this a business requirement to only retrieve records where the partId is specified?

Reply:
It's not required for fetching a courtAppearance by courtAppearanceId. However, the whole if block is related to updating a courtAppearance rather than updating the status. The ORDS endpoint calls a separate Justin procedure to do that which requires the partId as mandatory parameter to be passed so if we don't pass the partId that procedure will not run. That's why I included that partId check as well.

Reply:
Okay, that makes sense.